### PR TITLE
Fix stretched screen when entering and exiting fullscreen

### PR DESCRIPTION
**Pull request by alejx**
- [x] This change has been discussed with the author.
- [ ] It implements / fixes issue #.

### CODE REVIEW

File: iina/MPVController.swift [hunk 1]
```diff
@@ -1080,6 +1080,9 @@ fileprivate func mpvGetOpenGLFunc(_ ctx: UnsafeMutableRawPointer?, _ name: Unsaf
 
 fileprivate func mpvUpdateCallback(_ ctx: UnsafeMutableRawPointer?) {
   let layer = unsafeBitCast(ctx, to: ViewLayer.self)
```
Comment (saagarjha) [inline iina/MPVController.swift]: This isn't part of your changes, but I feel like `unsafeBitCast(_:to:)` is incorrect here. Perhaps this should use `UnsafeMutableRawPointer.assumingMemoryBound(to:)`?

File: iina/MPVController.swift [hunk 2]
```diff
@@ -1080,6 +1080,9 @@ fileprivate func mpvGetOpenGLFunc(_ ctx: UnsafeMutableRawPointer?, _ name: Unsaf
...
+  if layer.blocked {
```
Comment (saagarjha) [inline iina/MPVController.swift]: I'd use a guard here.

File: iina/MainWindowController.swift [hunk 3]
```diff
@@ -1440,31 +1432,24 @@ class MainWindowController: NSWindowController, NSWindowDelegate {
...
-      self.player.mpv.setFlag(MPVOption.Window.keepaspect, false)
-      DispatchQueue.main.sync {
-        for (_, constraint) in self.videoViewConstraints {
-          constraint.constant = 0
-        }
-        self.videoView.needsLayout = true
-        self.videoView.layoutSubtreeIfNeeded()
-        self.videoView.videoLayer.draw()
-      }
+    for (_, constraint) in self.videoViewConstraints {
```
Comment (saagarjha) [inline iina/MainWindowController.swift]: `for constraint in self.videoViewConstraints.values`

**Comment by lhc70000** [comment]
@saagarjha 

Fixed in dcc346f. An unmanaged reference of the VideoLayer was passed to mpv as the context, so here we should also use `Unmanaged` to load the pointer.

**Pull request merged**